lib/checkPullRequestLabels.js

```diff
@@ -363,3 +365,41 @@ module.exports.checkForIssueLabel = async function (context) {
     await context.github.issues.removeLabel(removeLabelParams);
   }
 };
+
+/**
+ * This function checks and prevent the removal of stale build label
+ * by unauthorised users.
+ *
+ * @param {import('probot').Context} context
+ */
+module.exports.checkStaleBuildLabelRemoval = async function (context) {
+  /**
+   * @type {import('probot').Octokit.IssuesGetLabelResponse} label
+   */
+  const label = context.payload.label;
+  const pullRequest = context.payload.pull_request;
+  if (label.name === OLD_BUILD_LABEL) {
+    const user = context.payload.sender;
+    const {data: lastCommit} = await context.github.repos.getCommit(
+      context.repo({
+        ref: pullRequest.head.sha
+      })
+    );
+
+    const lastCommitDate = new Date(lastCommit.commit.author.date);
+    // Commit is older than 2 days
+    if (MIN_BUILD_DATE > lastCommitDate) {
```

Comment:
Since we are making this check at multiple locations, this can be extracted into a utility function that accepts the pull request and context as parameters and returns a boolean value if the commit is older than the minimum build date.

Reply:
addressed in latest commit.